pkg/storage/manifest/raw/rawstorage.go

```diff
@@ -0,0 +1,152 @@
+package raw
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/weaveworks/ignite/pkg/storage"
+	"github.com/weaveworks/ignite/pkg/util"
+)
+
+type MappedRawStorage interface {
+	storage.RawStorage
+
+	AddMapping(key storage.Key, path string)
+	GetMapping(path string) (storage.Key, error)
+	RemoveMapping(key storage.Key)
+}
+
+func NewManifestRawStorage(dir string) MappedRawStorage {
```

Comment:
for clarity, let the struct implementation be called `mappedRawStorage` and the func be `NewMappedRawStorage` then

Reply:
in other words, go all-in on the Mapped word instead of manifest here

Reply:
:+1: it's now called `GenericMappedRawStorage` matching all our other (Raw)Storages implementing their interfaces.